Update phenotype import hive - e103

ima23 opened the pull request:
fixes:
- gwas NR
- add seq_region_ids update for human
- AnimalQTL: import date only updated for imported species, species requiring remapping are skipped
- update meta_coord phenotype_feature related entries

address previous PR comments/other fixes:
- use hash for branch job flows
- define Human_var , Human_gene run_types
- move run_type from global to internal param 
- make finish module standalone

jira tickets: ENSVAR-3223, ENSVAR-3455, ENSVAR-3373, ENSVAR-2724, ENSVAR-2871, ENSVAR-3140

helensch reviewed (commented):
Minor comments following initial review.
Review of pipeline flow being in second stage of review

ima23 commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm:
```diff
@@ -376,6 +376,36 @@ sub default_phenotype_class {
 #----------------------------
 # PUBLIC METHODS
 
+=head2 update_meta_coord
```
Yes, the method could be made more general and placed in a utils package. Let me know if you have suggestions of where, as it stands this is specific to this pipeline.

ima23 commented on modules/Bio/EnsEMBL/Variation/Utils/SeqRegionUtils.pm:
```diff
@@ -0,0 +1,106 @@
+
+=cut
+
+sub update_seq_region_ids {
```
Yes, the module is outside the pipeline and anyone can use it for any species.

helensch replied:
I was querying if the data was available for all species to use this sub.

ima23 commented on modules/Bio/EnsEMBL/Variation/Utils/SeqRegionUtils.pm:
```diff
@@ -0,0 +1,106 @@
+our @EXPORT_OK = qw(update_seq_region_ids);
+
+
+=head2 update_seq_region_ids
```
Good point. Will create a ticket for it.

ima23 commented:
Thanks @helensch  for checking. I've made the minor comment changes and made a ticket re SeqRegionUtils for the raised points.

helensch commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm:
```diff
@@ -43,20 +43,29 @@ use File::Path qw(make_path);
 my $count_ok = 1;
 
+my %groups_end_source = (
+  Human      => CGC,
```
Should this be HUMAN ?

helensch commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/Constants.pm:
```diff
@@ -54,37 +54,66 @@ our @EXPORT_OK = qw(
+use constant GROUP_RUN_TYPES => (ANIMALSET => ['OMIA','AnimalQTL'],
+                                MOUSE     => ['IMPC', 'MGI'],
+                                HUMAN     => ['GWAS', 'EGA',
+                                              'Orhpanet', 'MIMmorbid',
```
Typo in "Orhpanet"

helensch commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportHuman.pm:
```diff
@@ -67,24 +90,11 @@ sub write_output {
+      $self->dataflow_output_id($self->param('output_ids'), $source2branch{$run_type});
+      $self->print_pipelogFH( "Passing to $run_type import: ".scalar @{$self->param('output_ids')}." species\n") if $self->param('debug_mode');
+    } else {
+      $self->print_pipelogFH("Runtype $run_type not supproted!\n");
```
Typo in "supproted"
If run_type not supported,  should import continue?

helensch reviewed (commented):
Review of flow changes in PR done. Minor comments.
In future would be useful to review/document:
-  pipeline configuration: hive_capacity=1, empty input_ids, empty flow_into
- use of autoflow branch in runnables
- dataflow/branches in CheckPhenotypeAnnotation

ima23 commented:
Thanks for the feedback @helens. PR is ready for review again. I have added a few more fixes based on the production run. I have addressed the minor comments and created two tickets relating to SeqRegionUtils, ENSVAR-3659, and eHive questions, ENSVAR-3666. Please feel free to add to the tickets anything that would be useful for future discussions.

helensch commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/CheckPhenotypeAnnotation.pm:
```diff
@@ -105,58 +114,64 @@ sub run {
+
+      if ($fwd) {
+        #generally another import_source module
+        $self->dataflow_output_id($self->param('output_ids'), 2);
```
I now understand the module is controlling the flow to the different analysis. Suggest adding information on the what the branch flow is in the module.

helensch commented on modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm:
```diff
@@ -376,6 +376,36 @@ sub default_phenotype_class {
+  my $table_name ='phenotype_feature';
+
+  # delete previous phenotype_feature entries in update_meta_coord
+  my $mc_del_sth = $vdbh->do(qq[ DELETE FROM meta_coord WHERE table_name ='$table_name' ]);
```
The values returned by the do statements are not used. Variable suggests that it is a  statement handler.  do returns number of rows  or undef on error.

helensch reviewed (approved):
Thanks for the updates. I have added a ticket ENSVAR-3719 for suggestions for update_meta_coord.

helensch commented:
merged into master and release/103
